amd_openvx/openvx/hipvx/filter_kernels.cpp

```diff
@@ -206,7 +206,8 @@ int HipExec_Box_U8_U8_3x3(hipStream_t stream, vx_uint32 dstWidth, vx_uint32 dstH
     hipLaunchKernelGGL(Hip_Box_U8_U8_3x3, dim3(ceil((float)globalThreads_x/localThreads_x), ceil((float)globalThreads_y/localThreads_y)),
                         dim3(localThreads_x, localThreads_y), 0, stream, dstWidth, dstHeight, (uchar *)pHipDstImage, dstImageStrideInBytes,
                         (const uchar *)pHipSrcImage, srcImageStrideInBytes);
-
+    HIP_CHECK(hipGetLastError()); // Check for launch error
+    HIP_CHECK(hipDeviceSynchronize()); // Check for execution error
```

Comment:
hipDeviceSynchronize() shouldn't be called here. Please remove all the occurrences in this PR.

Reply:
@AryanSalmanpour how do we capture the execution errors on the kernels?

Reply:
@kiritigowda The execution errors will be captured by hipStreamSynchronize during the graph execution call. The calls to hipDeviceSynchronize in this context are incorrect because they force the execution of the hip kernels, which is not the intended behavior. Therefore, please remove all instances of hipDeviceSynchronize from this file.